examples/kitti/detectnet_network.prototxt

```diff
@@ -0,0 +1,2536 @@
+# DetectNet network
+
+# Data/Input layers
+name: "DetectNet"
+layer {
+  name: "train_raw_data"
+  type: "Data"
+  top: "raw_data"
+  data_param {
+    backend: LMDB
+    source: "examples/kitti/kitti_train_images.lmdb"
+    batch_size: 10
+  }
+  include: { phase: TRAIN }
+}
+layer {
+  name: "train_raw_label"
+  type: "Data"
+  top: "raw_label"
+  data_param {
+    backend: LMDB
+    source: "examples/kitti/kitti_train_labels.lmdb"
+    batch_size: 10
+  }
+  include: { phase: TRAIN }
+}
+layer {
+  name: "val_raw_data"
+  type: "Data"
+  top: "raw_data"
+  data_param {
+    backend: LMDB
+    source: "examples/kitti/kitti_test_images.lmdb"
+    batch_size: 6
+  }
+  include: { phase: TEST stage: "val" }
+}
+layer {
+  name: "val_raw_label"
+  type: "Data"
+  top: "raw_label"
+  data_param {
+    backend: LMDB
+    source: "examples/kitti/kitti_test_labels.lmdb"
+    batch_size: 6
+  }
+  include: { phase: TEST stage: "val" }
+}
+layer {
+  name: "deploy_data"
+  type: "Input"
+  top: "data"
+  input_param {
+    shape {
+      dim: 1
+      dim: 3
+      dim: 384
+      dim: 1248
+    }
+  }
+  include: { phase: TEST not_stage: "val" }
+}
+
+# Data transformation layers
+layer {
+  name: "train_transform"
+  type: "DetectNetTransformation"
+  bottom: "raw_data"
+  bottom: "raw_label"
+  top: "data"
+  top: "label"
+  detectnet_groundtruth_param: {
+    stride: 16
+    scale_cvg: 0.4
+    gridbox_type: GRIDBOX_MIN
+    coverage_type: RECTANGULAR
+    min_cvg_len: 20
+    obj_norm: true
+    image_size_x: 1248
+    image_size_y: 384
+    crop_bboxes: true
+  }
+   detectnet_augmentation_param: {
+    crop_prob: 1
+    shift_x: 32
+    shift_y: 32
+    flip_prob: 0.5
+    rotation_prob: 0
+    max_rotate_degree: 5
+    scale_prob: 0.4
+    scale_min: 0.8
+    scale_max: 1.2
+    hue_rotation_prob: 0.8
+    hue_rotation: 30
+    desaturation_prob: 0.8
+    desaturation_max: 0.8
+  }
+  transform_param: {
+    mean_value: 127
+  }
```

Comment:
> I was using a version of DIGITS that required the data layer to have a top called data but maybe you can do the shift in place now.

Yeah, that should work now that https://github.com/NVIDIA/DIGITS/pull/763 has been merged.

Depending on the implementation, maybe we can move the `transform_param` into the `Data` layer? Then for this Caffe example it would be clear that people should be subtracting the same value for deployment pipeline, and in DIGITS the mean value can be calculated automatically for each dataset?